futures-util/src/io/read_line.rs

```diff
@@ -29,20 +29,31 @@ impl<'a, R: AsyncBufRead + ?Sized + Unpin> ReadLine<'a, R> {
     }
 }
 
+pub(super) fn read_line_internal<R: AsyncBufRead + ?Sized>(
+    reader: Pin<&mut R>,
+    buf: &mut String,
+    bytes: &mut Vec<u8>,
+    read: &mut usize,
+    cx: &mut Context<'_>,
+) -> Poll<io::Result<usize>> {
+    let ret = ready!(read_until_internal(reader, b'\n', bytes, read, cx));
+    if str::from_utf8(&bytes).is_err() {
+        Poll::Ready(ret.and_then(|_| {
```

Comment:
nit: `map_err`?

Reply:
If that was `or_else`, we could use `map_err` instead (`and_then` does not touch the error.).

Reply:
Oh, sorry, I missed that this is actually the inversion-- it's `and_then`-ing and mapping the success value case to an error value. Ignore.